src/lib/containers/QueryWrapper.tsx

```diff
@@ -393,6 +403,9 @@ export default class QueryWrapper extends React.Component<
           })
         }
         const isFaceted = isFacetAvailable(data.facets)
+        if (this.props.onQueryResultBundleChange) {
+          this.props.onQueryResultBundleChange(JSON.stringify(data))
+        }
```

Comment:
This is quite the callback. Can't wait for us to have the chance to refactor this. Looks good